doc/rationale.md

```diff
@@ -0,0 +1,39 @@
+
+To this end Ignition only runs on the first boot of a machine. Being a general provisioning utility however, there's nothing stopping a user from using Ignition to set up some configuration management tool. This may even make a lot of sense depending on the organization and requirements the user is trying to meet.
+
+## Ignition produces the machine asked for, or no machine at all
+
+Ignition does what it needs to make the system match the state described in the Ignition config. If for any reason Ignition cannot deliver the exact machine that the config asked for, Ignition prevents the machine from booting successfully.
+
+For example if the user wanted a file from `https://example.com/foo.conf` to be downloaded and put on the disk and Ignition is unable to resolve the URL then Ignition will prevent the machine from successfully booting.
+
+## Ignition configs are declarative
+
+Ignition configs describe the state of a system. Ignition configs do _not_ list a series of steps that Ignition should take.
+
+To this end, Ignition configs do not allow users to provide arbitrary logic (including scripts for Ignition to run). Users describe which filesystems they want to exist, what files need to be created, what users should exist, and more. If a user wants more general setup ability, they can have Ignition set up a one-shot systemd unit that runs on first boot.
+
+## Ignition configs should not be written by hand
+
+Ignition configs are intended to not be human friendly, and users should not write Ignition configs by hand. Ignition configs are technically human readable, being JSON, but non-human-readable formats would have worked just as well.
+
+Hand writing Ignition configs encourages users to not validate configs before boot (which ensures wasting both user and machine time), JSON files aren't friendly with newlines (which makes writing configs by hand frustrating), and Ignition configs don't understand the particulars of the OS on which they'll be run (which requires users to have more understanding of the OS than they'd otherwise need).
+
+Instead a separate tool should be used to generate Ignition configs. In the case of Container Linux, files called Container Linux Configs can be written in YAML and then converted into Ignition configs. By having a Container Linux specific config schema and requiring users to go through a manual conversion process, all of the above pitfalls can be avoided.
+
+## Ignition runs in the initramfs
+
+When a Linux machine is booting it goes through a phase where the Linux kernel is running but the root filesystem is in memory and no real disks have been mounted yet.
+
+It is at this stage of the boot that Ignition runs. Because of this it can manipulate disks however it sees fit, even going so far as to reformat the root filesystem or set up software raid devices.
```

Comment:
"Ignition runs in a very early stage of the Linux boot process; before the network has started and before any disks have been mounted. Because of this, Ignition has the ability to fully configure disks, even going ..."